doc/user-manual/language/sort-system.lagda.rst

```diff
@@ -0,0 +1,112 @@
+..
+  ::
+  module sort.system where
+
+.. _sort-system:
+
+***********
+Sort System
+***********
+
+.. _intro-sorts:
+
+Agda's sorts system
+-------------------
+
+The implementation of Agda’s sorts system is closely based on the theory
+of pure type systems. Sorts (aka universes) are types whose members
+themselves are again types. The fundamental sort in Agda is named ``Set``
+and it denotes the universe of small types.
+
+But for some applications, other sorts are needed. Martin-Löf’s type theory
+had originally a rule ``Set : Set`` but Girard showed that it is inconsistent.
+This result is known as Girard’s paradox. Agda implements an
+:ref:`infinite hierarchy of universes <universe-levels>` that avoids this
+inconsistency.
+
+Under universe polymorphism, levels can be arbitrary terms, e.g., a
+level that contains free variables. Sometimes, we will have to check
+that some expression has a valid type without knowing what sort it has.
+
+funSort
+-------
+
+Agda’s internal representation of sorts implements a constructor (sort
+metavariable) representing an unknown sort. The constraint solver can
+compute these sort metavariables, just like it does when computing
+regular term metavariables.
+
+But this sort metavariable need other constructors to solve function
```

Comment:
need --> needs

The sentence is a bit unclear.

I think there should be a basic explanation that if in function type `A -> B` domain `A` has sort `sA` and codomain `B` has sort `sB`, then the function type has sort `funSort sA sB`. If `sA` and `sB` are known, `funSort sA sB` can be computed to a sort value.


Like there should be a basic explanation that if in dependent function type `(x : A) -> B` domain `A` has sort `sA` and codomain `B` has sort `sB` dependent on `x : A`, then the function type has sort `piSort sA (\ x -> sB)`.

Reply:
> need --> needs

Thanks, this happens when committing at night...

> I think there should be a basic explanation...

I agree, please see how it reads now.